internal/client/repository/badger/badger_repository.go

```diff
@@ -32,9 +35,13 @@ type Repository struct {
 	db      *badger.DB
 	metrics metrics
 
-	signatureMutexMap sync.Map // map[requestId]*sync.Mutex
-	proofsMutexMap    sync.Map // map[requestId]*sync.Mutex
-	valsetMutexMap    sync.Map // map[epoch]*sync.Mutex
+	signatureMutexMap sync.Map // map[requestId]*mutexWithUseTime
+	proofsMutexMap    sync.Map // map[requestId]*mutexWithUseTime
+	valsetMutexMap    sync.Map // map[epoch]*mutexWithUseTime
+
+	cleanupTicker     *time.Ticker
+	cleanupStop       chan struct{}
+	cleanupStaleAfter time.Duration
```

Comment:
we can probably get rid of these as they are only needed locally in the function. The channel is the only thing that external func needs to call.
```suggestion
	cleanupStop       chan struct{}
```